src/nsrt_learning.py

```diff
@@ -94,16 +94,25 @@ def segment_trajectory(trajectory: GroundAtomTrajectory) -> List[Segment]:
         # like Holding, Handempty, etc., because when doing symbolic planning,
         # we only have predicates, and not the continuous parameters that would
         # be used to distinguish between a PickPlace that is a pick vs a place.
-        if actions[t].has_option() and t < len(actions) - 1:
-            # We don't care about the last option in the traj because there's
-            # no next option after it, so we could never possibly want to
-            # segment due to that option. But we still need to check atoms here,
-            # because the length of all_atoms is one longer.
-            option_t = actions[t].get_option()
-            option_t1 = actions[t+1].get_option()
-            option_t_spec = (option_t.parent, option_t.objects)
-            option_t1_spec = (option_t1.parent, option_t1.objects)
-            if option_t_spec != option_t1_spec:
+        if actions[t].has_option():
+            # Check for a change in option specs.
+            if t < len(actions) - 1:
+                option_t = actions[t].get_option()
+                option_t1 = actions[t+1].get_option()
+                option_t_spec = (option_t.parent, option_t.objects)
+                option_t1_spec = (option_t1.parent, option_t1.objects)
+                if option_t_spec != option_t1_spec:
+                    switch = True
+            # Special case: assume that trajectory data never interrupts the
+            # final option (we will assert this). Under this assumption, we
+            # can safely segment without using any continuous info. Note that
+            # excluding the final option from the data is highly problematic
+            # when using demo+replay with the default 1 option per replay
+            # because all of the replay data would get excluded (unless there
```

Comment:
update this comment to make it more clear -- "because the replay data which causes no change in the symbolic state would get excluded."